services/storehost/storehost.go

```diff
@@ -1558,3 +1558,43 @@ func (t *StoreHost) RemoteReplicateExtent(tCtx thrift.Context, req *store.Remote
 
 	return nil
 }
+
+// ListExtents lists extents available on this storehost
+func (t *StoreHost) ListExtents(tCtx thrift.Context) (res *store.ListExtentsResult_, err error) {
+
+	t.m3Client.IncCounter(metrics.ListExtentsScope, metrics.StorageRequests)
+
+	sw := t.m3Client.StartTimer(metrics.ListExtentsScope, metrics.StorageLatencyTimer)
+	defer sw.Stop()
+
+	log := t.logger
+
+	log.Info("ListExtents()")
+
+	extents, err := t.xMgr.ListExtents()
```

Comment:
Let's follow the common pattern of error handling just for readability:

extents, err := t.xMgr.ListExtents()
if err != nil {
   return storeError
}

return result(extents)